applications/nrf5340_audio/src/audio/audio_datapath.c

```diff
@@ -762,12 +753,35 @@ void audio_datapath_just_in_time_check_and_adjust(uint32_t sdu_ref_us)
 	}
 }
 
+int audio_datapath_pres_delay_us_set(uint32_t delay_us)
+{
+	if (delay_us > CONFIG_AUDIO_MAX_PRES_DLY_US || delay_us < CONFIG_AUDIO_MIN_PRES_DLY_US) {
+		LOG_WRN("Presentation delay not supported: %d", delay_us);
+		return -EINVAL;
+	}
+
+	ctrl_blk.pres_comp.pres_delay_us = delay_us;
+
+	LOG_DBG("Presentation delay set to %d us", delay_us);
+
+	return 0;
+}
+
+void audio_datapath_pres_delay_us_get(uint32_t *delay_us)
+{
+	*delay_us = ctrl_blk.pres_comp.pres_delay_us;
+}
+
 void audio_datapath_sdu_ref_update(uint32_t sdu_ref_us)
 {
-	if (ctrl_blk.stream_started) {
-		ctrl_blk.previous_sdu_ref_us = sdu_ref_us;
-	} else {
-		LOG_WRN("Stream not startet - Can not update sdu_ref_us");
+	if (IS_ENABLED(CONFIG_AUDIO_SOURCE_I2S)) {
+		if (ctrl_blk.stream_started) {
+			ctrl_blk.previous_sdu_ref_us = sdu_ref_us;
```

Comment:
SDU ref is not updated twice the amount that is necessary if BIDIR (both when sending and receiving).
I guess we do not want audio_datapath to know whether it is BIDIR or not?

Reply:
I probably looked at the BIS gateway code instead of CIS gateway code